src/server.c

```diff
@@ -2598,17 +2621,22 @@ void initServer(void) {
 
     /* Create the Redis databases, and initialize other internal state. */
     for (j = 0; j < server.dbnum; j++) {
-        server.db[j].dict = dictCreate(&dbDictType);
+        int slotCount = (server.cluster_enabled) ? CLUSTER_SLOTS : 1;
+        server.db[j].dict = dictCreateMultiple(&dbDictType, slotCount);
         server.db[j].expires = dictCreate(&dbExpiresDictType);
```

Comment:
Are you planning to also make the `expire` dictionary to be per slot? 

It is pretty much a mirror image of the main dictionary. If you want per slot then you should also do the same for `expire` dictionary

Reply:
To get all the keys in a slot, we don't need one expire dict per slot.

Reply:
There is no benefit in splitting expiry dictionary per slot as it doesn't have any memory overhead in cluster mode.

Reply:
OK makes sense.